vendor/lib/src/dependencies.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:pubspec_parse/pubspec_parse.dart';
+import 'package:vendor/src/config.dart';
+import 'package:vendor/src/context.dart';
+
+/// Check dependencies for all vendored packages.
+///
+/// If a dependency of a vendored package must be:
+///  * satisfied by the dependency constrant in [rootPubspec],
+///  * declared in `dependency_overrides ` in [rootPubspec], or,
+///  * rewritten to another vendored package.
+///
+/// If neither of these scenarios is the case, then this method will print a
+/// warning with hints of how to resolve situation.
```

Comment:
Do you really give any hints?

Reply:
Not yet, I guess...  I wanted to suggest users could call `dart pub add`